test/test_sync_subscriptions.cpp

```diff
@@ -389,4 +389,19 @@ TEST(Sync_SubscriptionStoreNotifications)
     CHECK_EQUAL(std::move(fut).get(), SubscriptionSet::State::Complete);
 }
 
+TEST(Sync_RefreshSubscriptionSetInvalidSubscriptionStore)
+{
+    SHARED_GROUP_TEST_PATH(sub_store_path)
+    SubscriptionStoreFixture fixture(sub_store_path);
+    auto store = SubscriptionStore::create(fixture.db, [](int64_t) {});
+    // Because there are no subscription sets yet, get_latest should point to an empty object
+    auto latest = std::make_unique<SubscriptionSet>(store->get_latest());
+    CHECK(latest->begin() == latest->end());
+    // The SubscriptionStore gets destroyed.
+    store.reset();
+
+    // Throws since the SubscriptionStore is gone.
+    CHECK_THROW_ANY(latest->refresh());
```

Comment:
Can we check that it throws a std::logic_error? I think this could throw for other reasons.